cl/phase1/core/state/accessors.go

```diff
@@ -277,22 +338,25 @@ func ExpectedWithdrawals(b abstract.BeaconState, currentEpoch uint64) []*cltypes
 			}
 			withdrawals = append(withdrawals, newWithdrawal)
 			nextWithdrawalIndex++
-		} else if isPartiallyWithdrawableValidator(b.BeaconConfig(), currentValidator, currentBalance) { // Check if the validator is partially withdrawable
+		} else if isPartiallyWithdrawableValidator(b, currentValidator, currentBalance) { // Check if the validator is partially withdrawable
 			// Add a new withdrawal with the validator's withdrawal credentials and balance minus the maximum effective balance
+			amount := currentBalance - b.BeaconConfig().MaxEffectiveBalance
+			if b.Version() >= clparams.ElectraVersion {
+				amount = currentBalance - getMaxEffectiveBalanceElectra(currentValidator, b.BeaconConfig())
```

Comment:
same as above

Reply:
The same, `isPartiallyWithdrawableValidator` helps to check underflow is impossible